### DISCUSSION TITLE
Help with Jukebox mode

### DISCUSSION THREAD

**Discussion by gapato** [navidrome/navidrome#2646] [Q&A] [answered]
I am trying to setup Navidrome's Jukebox mode (thank you for that) on a headless server at home following the documentation [here](https://www.navidrome.org/docs/usage/jukebox/), but when playing a song, the audio from the browser, it plays on the client's speaker, not on the server's output.

I've installed mpv and selected an output device, and checked that mpv correctly played files on that device. I've also configured navidrome like that:

    Jukebox.Enabled = true
    Jukebox.Default = "pulse"
    
    Jukebox.Devices = [
        [ "pulse",          "pulse/alsa_output.pci-0000_00_08.0.analog-stereo" ]
    ]

Navidrome output when starting seems fine:

    ...
    INFO[0000] Starting playback server                     
    INFO[0000] Login rate limit set                          requestLimit=5 windowLength=20s
    INFO[0000] Found ffmpeg                                  path=/usr/bin/ffmpeg          
    INFO[0000] 1 audio devices found                        
    INFO[0000] Using audio device: pulse/alsa_output.pci-0000_00_08.0.analog-stereo 
    ...

But as mentioned, the audio plays on the client, not the server. After starting the server, there is no mpv process running (checked with `pgrep mpv`), is that normal? Is there anything I should do in the browser to enable jukebox mode?

    $ navidrome --version
    0.50.0 (fee0f40a)

    $ mpv --version
    mpv 0.32.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
     built on UNKNOWN
    ffmpeg library versions:
       libavutil       56.51.100
       libavcodec      58.91.100
       libavformat     58.45.100
       libswscale      5.7.100
       libavfilter     7.85.100
       libswresample   3.7.100
    ffmpeg version: 4.3.6-0+deb11u1

**Comment by dweymouth** [comment, marked as answer]
You need to use a Subsonic client like DSub or play:Sub that supports the Jukebox mode. It's a UI setting that has to be turned on in the client to tell Navidrome to play the file using the Jukebox output instead of streaming it to the client. Jukebox mode support has not been added to the builtin web UI.

**Reply by gapato** [reply]
I see! It took me a while to find how to do it with DSub but it works now. Thank you.

**Reply by ghost** [reply]
iSub is another one that has a jukebox mode.

**Reply by nickolay** [reply]
iSub worked for me [eventually](https://github.com/navidrome/navidrome/issues/2771#issuecomment-2089152259), but not play:Sub...

**Reply by pruperting** [reply]
Any way a non git user can help getting this added to the documentation, as this really wasn't clear to me from the docs - my impression was that if you enabled this in the navidrome config, it would just work when I pressed play in the webUI!

**Reply by deluan** [reply]
@pruperting Yeah, I agree this should be in the docs. You can try editing the page yourself by using this:
![Screenshot 2024-09-11 at 8 29 09 PM](https://github.com/user-attachments/assets/c4840770-93df-4e3c-81ea-937e592a09b8)

It will open the page in the GH editor, and it should be very straightforward. If you have questions or feel it is not that simple, post here and I will carve some time to do it.

**Reply by shauncksm** [reply]
I've created a pull request [here](https://github.com/navidrome/website/pull/177). My very first pull request in fact 😊

**Comment by wie-was** [comment]
I was literally struggling with the same issue yesterday :) It's great to see that the documentation is now more clear about this.

Using DSub, as suggested, I found out that there is yet another problem on my system. I don't want to hijack this thread, but it seems quite fitting. I'm posting the log of what happens when I'm trying to send music to the "Subsonic Jukebox" in DSub. 

```
volumio@volumio:~$ journalctl --unit=navidrome.service -n 10 --no-pager
-- Logs begin at Sun 2024-09-15 17:48:19 CEST, end at Mon 2024-09-16 21:36:11 CEST. --
Sep 16 21:36:06 volumio navidrome[4105]: time="2024-09-16T21:36:06+02:00" level=info msg="JukeboxControl request received" action=status requestId=volumio/fFkJZzaS7D-000539
Sep 16 21:36:06 volumio navidrome[4105]: time="2024-09-16T21:36:06+02:00" level=debug msg="processing Status action on: Name: SA9023, Gain: 1.0000, Loaded track: %!s(<nil>), queue: #Items: 12, idx: 11, files: 0:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/01 The Package.mp3 1:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/02 Weak And Powerless.mp3 2:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/03 The Noose.mp3 3:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/04 Blue.mp3 4:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/05 Vanishing.mp3 5:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/06 A Stranger.mp3 6:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/07 The Outsider.mp3 7:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/08 Crimes.mp3 8:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/09 The Nurse Who Loved Me.mp3 9:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/10 Pet.mp3 10:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/11 Lullaby.mp3 11:/media/Lacie_R_1TB/Music collection/Alben/A Perfect  Circle - Thirteenth Step/12 Gravity.mp3 " requestId=volumio/fFkJZzaS7D-000539
Sep 16 21:36:06 volumio navidrome[4105]: time="2024-09-16T21:36:06+02:00" level=debug msg="API: Successful response" endpoint=/rest/jukeboxControl.view requestId=volumio/fFkJZzaS7D-000539 status=OK
Sep 16 21:36:06 volumio navidrome[4105]: time="2024-09-16T21:36:06+02:00" level=debug msg="HTTP: GET http://volumio.freeddns.org:4533/rest/jukeboxControl.view?u=raphael&s=[REDACTED]&t=[REDACTED]&v=1.2.0&c=DSub&action=status" elapsedTime=7ms httpStatus=200 remoteAddr="192.168.192.1:38432" requestId=volumio/fFkJZzaS7D-000539 responseSize=249 userAgent=DSub
Sep 16 21:36:11 volumio navidrome[4105]: time="2024-09-16T21:36:11+02:00" level=debug msg="API: New request /rest/jukeboxControl.view" client=DSub requestId=volumio/fFkJZzaS7D-000540 username=raphael version=1.2.0
Sep 16 21:36:11 volumio navidrome[4105]: time="2024-09-16T21:36:11+02:00" level=debug msg="Processing GetDevice" user=raphael
```

Audio works fine apart from that, eg. `mpv --audio-device alsa/sysdefault:CARD=Audio /media/Lacie_R_1TB/Music\ collection/Alben/A\ Perfect\ \ Circle\ -\ Thirteenth\ Step/01\ The\ Package.mp3` plays the song. Also [Volumio](https://volumio.com/get-started/) is running on the device and works fine.

Trying to play random other songs with the command above, mpv crashes and the whole shell freezes, last message being `* failed to open vchiq instance`

Also, I never see `INFO[0000] Starting playback server` in the log.

Something with mpv seems to be off.

OS is Volumio OS / Raspbian GNU/Linux 10 (buster) Linux 6.1.69-v7+ on a Raspi 3B.

Maybe someone has encountered something similar - I'm investigating this further in any case. So happy with my neat little navidrome setup :)

Oh, and Volumio (volusonic plugin) is working fine for playing audio from the navidrome-server (on the same device, hehe). The only small but critical downside is that the volusonic plugin apparently does not feedback the navidrome server any playing statistics, nor can you rate songs.

**Reply by pruperting** [reply]
I've a similar issue, reported here: https://github.com/navidrome/navidrome/issues/3256. I suspect this is down to some strange combination of permissions and config settings across navidrome, mpv and the user.
